### ISSUE TITLE
GDScript: Blank/comment script lines use up CPU cycles inside for loop

### ISSUE THREAD

**Issue by wyattbiker** [godotengine/godot#33089]
**Godot version:**
3.2.alpha2 IDE
**OS/device including version:**
Mac OS X, Mojave

**Issue description:**
GDScript: Blank/comment script lines use up CPU cycles inside for loop. Seems to be  treated like a `pass` statement. Also the debugger single steps into comments/blank lines (instead of over) which is why I was wondering of the impact.

**Steps to reproduce:**
Create a Control scene and add script below. Test with blank lines and without blank lines.
For one blank line 1000 iterations on my system I get 185μs vs 148μs. Additional blank/comment lines you add the greater the cpu time.

```gdscript
extends Control
func _ready() -> void:
	var end_time:int
	var x:int = 0
	var start_time:int=OS.get_ticks_usec()
	for i in range(1000):
		
		x += 1
	end_time=OS.get_ticks_usec() 
	print("Elapsed μs: ", end_time - start_time)
	pass # Replace with function body.
```

**Comment by Calinou** [comment]
Does this occur in a project exported in release mode?

**Comment by wyattbiker** [comment]
I export to a .dmg file which is the installer format for Mac OS X. I install it and run. It has same behaviour. ie CPU cycles are consumed on blank/comments.

Here are the settings. I don't see a specific Release Mode flag.
![image](https://user-images.githubusercontent.com/930478/67622758-5d86c780-f7eb-11e9-9c12-70582cde0fc2.png)

**Comment by bruvzg** [comment]
> Does this slowdown occur in a project exported in release mode?

I can't reproduce it with `release` build, only with `debug`, here are results (average of 5000 runs with 1000 iterations each):

| Extra Lines | Release Time | Debug Time |
| ----------- | ------------ | ---------- |
| 0           | 53 μs        | 231 μs     |
| 1           | 52 μs        | 285 μs     |
| 10          | 52 μs        | 525 μs     |
| 100         | 52 μs        | 3011 μs    |

**Comment by bruvzg** [comment]
> I don't see a specific Release Mode flag.

It's in the corner of file selection dialogue, after you click `Export Project`.
![Screenshot 2019-10-26 at 19 47 29](https://user-images.githubusercontent.com/7645683/67623058-8fb71a00-f829-11e9-91cb-c8081a5b9b33.png)

**Comment by wyattbiker** [comment]
But yes, blank lines do not take up CPU time when Debug Flag is unchecked. Not as serious, but would be worthwhile pointing out that blanks and comments are treated as '`pass`' during dev.

Irrelevant comment: Didn't see that flag on the fileopen dialog. Strange place for it in my opinion.

**Comment by Calinou** [comment]
> Irrelevant comment: Didn't see that flag on the fileopen dialog. Strange place for it in my opinion.

Feel free to suggest improvements in #20219 :slightly_smiling_face:

**Comment by GoDino** [comment]
I believe a debugger should not step into blank lines/comments at all. With a small change this could be achieved, which should fix the bug. Would that be ok?
Already have a working version that does this, would be happy to make a PR.

**Comment by Calinou** [comment]
@GoDino Feel free to open a pull request for this :slightly_smiling_face:

**Comment by KoBeWi** [comment]
Fixed in 4.0
